src/components/weatherCard/weatherCard.js

```diff
@@ -26,28 +37,55 @@ export const WeatherCard = () => {
           "," +
           position.coords.longitude
       );
-
       const rawWeatherData = await response.json();
       weatherDispatch(actionHandler(rawWeatherData));
+      console.log("Ready");
     } catch (error) {
       console.log("error", error);
     }
   };
+
   return (
     <>
-      <h1>Weather</h1>
-      <button onClick={getWeather}>click me</button>
       <WeatherCardWrapper>
-        {weather.weatherData === null && !weather.isLoading  ? (
-          <p>click to load data </p>
-        ) : weather.isLoading && weather.weatherData === null ? (
-          <Spinner />
-        ) : (
-          <>
-            <p>Time is {weather.weatherData.time}</p>
-            <p>Temp is {weather.weatherData.temperature} deg</p>
-          </>
-        )}
+        <Card
+          bg="info"
+          border="dark"
+          text="light"
+          style={{ width: "30vw" }}
+          onClick={() => {
+            getWeather();
+            triggerCollapse();
+          }}
+        >
+          <div style={{ maxWidth: "500px" }}>
+            {weather.weatherData === null ? <p>Click to get weather</p> : null}
+            <Collapse in={open}>
+              {weather.weatherData === null && !weather.isLoading ? (
+                <div> </div>
+              ) : weather.isLoading && weather.weatherData === null ? (
+                <div>
+                  <Spinner animation="border" role="status" />
+                </div>
+              ) : (
+                <div>
+                  <p>Time is {weather.weatherData.time}</p>
+                  <p>Temp is {weather.weatherData.temperature} deg</p>
+                  <p>Humidity is {weather.weatherData.humidity}</p>
+                  <p>
+                    Now it is{" "}
+                    {weather.weatherData.isDay === "yes" ? "daytime" : "night"}{" "}
```

Comment:
i think you need to convert isDay to boolean in action or store , because its bullshit